server/workflow/workflow_server.go

```diff
@@ -321,24 +365,60 @@ func (s *workflowServer) PodLogs(req *workflowpkg.WorkflowLogRequest, ws workflo
 	ctx := ws.Context()
 	wfClient := auth.GetWfClient(ctx)
 	kubeClient := auth.GetKubeClient(ctx)
-	_, err := s.getWorkflowAndValidate(ctx, req.Namespace, req.Name, metav1.GetOptions{})
+	wf, err := s.getWorkflow(ctx, req.Namespace, req.Name, metav1.GetOptions{})
 	if err != nil {
 		return err
 	}
+	err = s.validateWorkflow(wf)
+	if err != nil {
+		return err
+	}
+	req.Name = wf.Name
 	return logs.WorkflowLogs(ctx, wfClient, kubeClient, req, ws)
 }
 
-func (s *workflowServer) getWorkflowAndValidate(ctx context.Context, namespace string, name string, options metav1.GetOptions) (*v1alpha1.Workflow, error) {
+func (s *workflowServer) getWorkflow(ctx context.Context, namespace string, name string, options metav1.GetOptions) (*v1alpha1.Workflow, error) {
+	const LATEST_ALIAS = "@latest"
+	if name == LATEST_ALIAS {
+		latest, err := getLatestWorkflow(ctx, namespace)
+		if err != nil {
+			return nil, err
+		} else {
+			fmt.Printf("Resolved alias %s to workflow %s.\n", LATEST_ALIAS, latest.Name)
```

Comment:
Please use `log.Infof` for this. As a rule of thumb, you should almost never use `fmt.Printf` for non-debugging uses
```suggestion
			log.Infof("Resolved alias %s to workflow %s.\n", LATEST_ALIAS, latest.Name)
```

Reply:
Also, the else statement here is unnecessary since `if err != nil {`, we return